External/ietl/ietl/eigeninterface.h

```diff
@@ -44,7 +44,7 @@ void mult(const Eigen::SparseMatrix<TCoeffs>& a,
 template <class matrix_t, class TCoeffs>
 void mult(const matrix_t& a, const Eigen::Matrix<TCoeffs, Eigen::Dynamic, 1>& x,
           Eigen::Matrix<TCoeffs, Eigen::Dynamic, 1>& y) {
-  y = a.Apply(x);
+  y = a(x);
```

Comment:
What's wrong with the method Apply? The operator () is really not suitable to indicate the intent of a multiplication. If you want to move from a method to an operator, I think this should be the * operator !

Reply:
Changes to this particular file are temporary (I am working on removing dependencies on IETL altogether). `mult` is a template which we instantiate with `Eigen::MatrixXcd`, `Eigen::SparseMatrix<Complex>`, and `netket::AbstractOperator`. In such cases it's pretty standard to pass function objects around and they implement `operator()` rather than `Apply`.

Reply:
I do not agree that this is a good naming, but it seems that you have already merged this

Reply:
@gcarleo Sorry. I merged the PR despite this point being still open since I was expecting this code to be removed anyway in #232.